support/arr/arr.go

```diff
@@ -0,0 +1,633 @@
+		} else if i < len(arr)-1 {
+			builder.WriteString(delimiter)
+		}
+	}
+	return builder.String()
+}
+
+// KeyBy Key an associative array by a field or using a callback.
+func KeyBy[T any, K comparable](array []T, key K) ([]T, error) {
+	return nil, ErrNoImplementation
+}
+
+// PrependKeysWith Prepend the key names of an associative array.
+func PrependKeysWith[T any](array []T, prependWith string) ([]T, error) {
+	return nil, ErrNoImplementation
+}
+
+// Only returns a subset of the items from the given map with specified keys.
+func Only[T any](arr []T, keys any) []T {
+	var res []T
+	switch v := keys.(type) {
+	case int:
+		if v < 0 || v > len(arr)-1 {
+			return res
+		}
+		res = append(res, arr[v])
+
+	case []int:
+		for _, vv := range v {
+			if vv < 0 || vv > len(arr)-1 {
+				return res
+			}
+			res = append(res, arr[vv])
+		}
+	}
+	return res
+}
+
+// Pluck an array of values from an array.
+func Pluck[T any](array []T, value T, key ...int) ([]T, error) {
+	return nil, ErrNoImplementation
+}
+
+// ExplodePluckParameters Explode the "value" and "key" arguments passed to "pluck".
+func ExplodePluckParameters[T any](array []T, key ...int) ([]T, error) {
+	return nil, ErrNoImplementation
+}
+
+// Map Run a map over each of the items in the array.
+func Map[T, U any](arr []T, fn func(T, int) U) []U {
+	res := make([]U, len(arr))
+	for i, v := range arr {
+		res[i] = fn(v, i)
+	}
+	return res
+}
+
+// Prepend the given value to the beginning of an array or associative array.
+func Prepend[T any](arr []T, value T) []T {
+	return append([]T{value}, arr...)
+}
+
+// Pull Get a value from the array, and remove it.
+func Pull[T any](arr []T, key int, def T) ([]T, T, error) {
+	v, _ := Get(arr, key, def)
+
+	res, err := Forget(arr, key)
+	return res, v, err
+}
+
+// Query Convert the array into a query string.
+func Query[T any](m []T) (*string, error) {
+	return nil, ErrNoImplementation
+}
+
+// Random returns one or a specified number of random values from a slice.
+func Random[T any](arr []T, number *int) ([]T, error) {
+	//rand.Seed(time.Now().UnixNano())
+
+	requested := 1
+	if number != nil {
+		requested = *number
+	}
+
+	count := len(arr)
+
+	if requested > count {
+		return nil, ErrExceedMaxLength
+	}
+
+	if number == nil {
+		return []T{arr[rand.Intn(count)]}, nil
+	}
+
+	if requested == 0 {
+		return []T{}, nil
+	}
+
+	indices := rand.Perm(count)[:requested]
+
+	res := make([]T, 0, requested)
+	for _, index := range indices {
+		res = append(res, arr[index])
+	}
+
+	return res, nil
+}
+
+// Set an array item to a given value using int key
+// todo: thread safe?
+func Set[T any](arr *[]T, key int, value T) error {
+	if key < 0 {
+		return ErrInvalidKey
+	}
+
+	if key >= len(*arr) {
+		newSlice := make([]T, key+1)
+		copy(newSlice, *arr)
+		*arr = newSlice
+	}
+
+	(*arr)[key] = value
+	return nil
+}
+
+// Shuffle the given array and return the result.
+func Shuffle[T any](arr []T, seed *int64) []T {
```

Comment:
Q: Can we remove the seed parameter?
```suggestion
func Shuffle[T any](arr []T) []T {
```